Add linting grunt tasks

justinvdm opened the pull request:
This resolves #2. For some reason, getting 307's when I use hub.

justinvdm committed:
Add jscs linting task

justinvdm committed:
Add jshint linting

justinvdm committed:
Add a basic travis set up

justinvdm committed:
jshint fix

justinvdm commented:
Ready for review. Will add scss linting in a separate PR, it probably involves doing ruby things in our travis config if we use https://github.com/ahmednuaman/grunt-scss-lint.

hodgestar commented on gruntfile.js:
```diff
@@ -1,11 +1,26 @@
+      js: ['gruntfile.js']
+    },
+    scsslint: {
+      all: ['<%= paths.scss %>']
```
Should we add an empty `paths: scss: []`?

justinvdm replied:
hrm, `scsslint` shouldn't be here, will remove.

hodgestar commented:
Left one minor comment, otherwise looks good.

justinvdm committed:
Remove unused scsslint grunt task config

SteveBarnett commented on .jshintrc:
```diff
@@ -0,0 +1,20 @@
+  "unused": "vars",
+  "bitwise": true,
+  "eqeqeq": true,
+  "es3": true,
```
After discussion with @justinvdm, we've decided to bump this to `es5`. This will also let use change our `preset` to [airbnb](https://github.com/jscs-dev/node-jscs/blob/master/presets/airbnb.json), which is (and I quote): "nice."

We need to be careful of IE and trailing commas, though.

SteveBarnett commented:
Left a comment about our discussion, otherwise :+1: .

justinvdm committed:
Fix jshint grunt task to actually look at .jshintrc

justinvdm committed:
Adjust .jshintrc

justinvdm commented:
Ready for re-review. Turns out the grunt jshint task wasn't looking at `.jshintrc`, fixed that and then realised a few things about the .jshintrc. I've removed some jshint rules, will make comments about why.

justinvdm committed:
Bring back jshint undef rules

justinvdm committed:
Remove duplicates in .jshintrc

justinvdm committed:
Move to airbnb jscs preset

justinvdm commented:
.jshintrc changes and why:
  - Removed `"maxstatements": 4`: This isn't practical. For example, it complains about our gruntfile because of `module.exports = function(grunt) { ... }`, even thought that's a common case.
- Added `"laxbreak": true` so we can do write conditionals in an (arguably) more readable way:
```javascript
function foo(bar) {
  return bar
    ? 23
    : 42
}
```

The second change is probably less important than the first, though it is quite important to me :)

hodgestar commented:
:+1:

Is `iojs` important? From Wikipedia:
```
As of the week of May 15, 2015, the io.js organization has voted and officially agreed to merge back with the Node.js project under the rubric of a new foundation, the Node Foundation.[50] The combined organization will be named 'nodejs'.
```

justinvdm commented:
@hodgestar Not very, happy to remove it if you'd prefer, though I think it might make sense to keep it around while some libraries are still only expecting io. Not sure how likely that is. [jsdom](https://github.com/tmpvar/jsdom) is one example, though I doubt we will be using it.

hodgestar commented:
Happy for it to stick around. :)

SteveBarnett commented:
Dig the airbnb jscs.

I don't agree with `laxbreak`, but: ¯\_(ツ)_/¯ I won't be using it, so: :+1: .
